sdk/analytics.ts

```diff
@@ -1,32 +1,52 @@
 import { pino, Logger } from 'pino';
 import { Fetch } from "./types.js";
+import { Flags } from "./models.js";
 
-const ANALYTICS_ENDPOINT = 'analytics/flags/';
+export const ANALYTICS_ENDPOINT = './analytics/flags/';
 
-// Used to control how often we send data(in seconds)
+/** Duration in seconds to wait before trying to flush collected data after {@link trackFeature} is called. **/
 const ANALYTICS_TIMER = 10;
 
+const DEFAULT_REQUEST_TIMEOUT_MS = 3000
+
+export interface AnalyticsProcessorOptions {
+    /** URL of the Flagsmith analytics events API endpoint
+     * @example https://flagsmith.example.com/api/v1/analytics
+     */
+    analyticsUrl?: string;
+    /** Client-side key of the environment that analytics will be recorded for. **/
+    environmentKey: string;
+    /** Duration in milliseconds to wait for API requests to complete before timing out. Defaults to {@link DEFAULT_REQUEST_TIMEOUT_MS}. **/
+    requestTimeoutMs?: number;
+    logger?: Logger;
+    /** Custom {@link fetch} implementation to use for API requests. **/
+    fetch?: Fetch
+    
+    /** @deprecated Use {@link analyticsUrl} instead. **/
+    baseApiUrl?: string;
```

Comment:
If we've never exposed this, we can just remove it rather than deprecate it, right?

Reply:
Sorry, I could've been clearer in the PR description. We are currently exporting `AnalyticsProcessor`, so this would be a breaking change if we removed it. The only new export from this PR is `AnalyticsProcessorOptions`, which is just an explicit type definition for the constructor parameters.

Reply:
For reference https://github.com/Flagsmith/flagsmith-nodejs-client/pull/168/files/199532eb83b48d217a8544ce3aa29dab90e0bfa4#diff-66fd8b30e073dfc58786b90e78803df61c90731cd651e24cce5163022f08e43fL20